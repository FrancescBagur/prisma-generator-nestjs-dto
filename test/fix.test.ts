import { testFixtures } from './fixtures';

testFixtures('fixtures/basic/**/*.prisma');
