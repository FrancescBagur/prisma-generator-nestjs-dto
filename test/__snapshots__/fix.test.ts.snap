// Vitest Snapshot v1

exports[`fixtures/basic/**/*.prisma 1`] = `
[
  {
    "content": "
  export class ConnectProductDto {
  id: string;
  }
  ",
    "fileName": "../src/@generated/prisma-generator-nestjs-dto/connect-product.dto.ts",
  },
  {
    "content": "
import {Prisma} from '@prisma/client'
import {ApiExtraModels} from '@nestjs/swagger'
import {ConnectCategoryDto} from './connect-category.dto'
import {ConnectCompanyDto} from './connect-company.dto'
export class CreateProductCategoryRelationInputDto {
      connect: ConnectCategoryDto;
  }
export class CreateProductCompanyRelationInputDto {
      connect: ConnectCompanyDto;
  }
@ApiExtraModels(ConnectCategoryDto,CreateProductCategoryRelationInputDto,ConnectCompanyDto,CreateProductCompanyRelationInputDto)
export class CreateProductDto {
  name: string;
  description: string;
  images: Prisma.InputJsonValue[];
  highlighted?: boolean;
  reviewCount?: number;
  attributes?: Prisma.InputJsonValue;
  score?: number;
  categoryId: string;
  companyId: string;
  category: CreateProductCategoryRelationInputDto;
  company: CreateProductCompanyRelationInputDto;
  createdAt?: Date;
  updatedAt?: Date;
}
",
    "fileName": "../src/@generated/prisma-generator-nestjs-dto/create-product.dto.ts",
  },
  {
    "content": "
import {Prisma} from '@prisma/client'


export class UpdateProductDto {
  id: string ;
  name?: string ;
  description?: string ;
  images?: Prisma.JsonValue[] ;
  highlighted?: boolean  | null;
  reviewCount?: number  | null;
  attributes?: Prisma.JsonValue  | null;
  score?: number  | null;
}
",
    "fileName": "../src/@generated/prisma-generator-nestjs-dto/update-product.dto.ts",
  },
  {
    "content": "
import {Prisma} from '@prisma/client'
import {Category} from './category.entity'
import {Company} from './company.entity'
import {IntersectionType} from '@nestjs/swagger'

export class Product {
  id: string ;
  name: string ;
  description: string ;
  images: Prisma.JsonValue[] ;
  highlighted: boolean  | null;
  reviewCount: number  | null;
  attributes: Prisma.JsonValue  | null;
  score: number  | null;
  categoryId: string ;
  companyId: string ;
}
export class ProductRel {
  category: Category ;
  company: Company ;
}
export class ProductFull extends IntersectionType(Product, ProductRel) {}
",
    "fileName": "../src/@generated/prisma-generator-nestjs-dto/product.entity.ts",
  },
  {
    "content": "
  export class ConnectCategoryDto {
  id: string;
  }
  ",
    "fileName": "../src/@generated/prisma-generator-nestjs-dto/connect-category.dto.ts",
  },
  {
    "content": "



export class CreateCategoryDto {
  
}
",
    "fileName": "../src/@generated/prisma-generator-nestjs-dto/create-category.dto.ts",
  },
  {
    "content": "



export class UpdateCategoryDto {
  id: string ;
}
",
    "fileName": "../src/@generated/prisma-generator-nestjs-dto/update-category.dto.ts",
  },
  {
    "content": "
import {Product} from './product.entity'
import {IntersectionType} from '@nestjs/swagger'

export class Category {
  id: string ;
}
export class CategoryRel {
  Product: Product[] ;
}
export class CategoryFull extends IntersectionType(Category, CategoryRel) {}
",
    "fileName": "../src/@generated/prisma-generator-nestjs-dto/category.entity.ts",
  },
  {
    "content": "
  export class ConnectCompanyDto {
  id: string;
  }
  ",
    "fileName": "../src/@generated/prisma-generator-nestjs-dto/connect-company.dto.ts",
  },
  {
    "content": "



export class CreateCompanyDto {
  
}
",
    "fileName": "../src/@generated/prisma-generator-nestjs-dto/create-company.dto.ts",
  },
  {
    "content": "



export class UpdateCompanyDto {
  id: string ;
}
",
    "fileName": "../src/@generated/prisma-generator-nestjs-dto/update-company.dto.ts",
  },
  {
    "content": "
import {Product} from './product.entity'
import {IntersectionType} from '@nestjs/swagger'

export class Company {
  id: string ;
}
export class CompanyRel {
  Product: Product[] ;
}
export class CompanyFull extends IntersectionType(Company, CompanyRel) {}
",
    "fileName": "../src/@generated/prisma-generator-nestjs-dto/company.entity.ts",
  },
  {
    "content": "
  export class ConnectAnotherDto {
  id1: string;
  id2: string;
  }
  ",
    "fileName": "../src/@generated/prisma-generator-nestjs-dto/connect-another.dto.ts",
  },
  {
    "content": "



export class CreateAnotherDto {
  id2: string;
}
",
    "fileName": "../src/@generated/prisma-generator-nestjs-dto/create-another.dto.ts",
  },
  {
    "content": "



export class UpdateAnotherDto {
  id1: string ;
  id2: string ;
}
",
    "fileName": "../src/@generated/prisma-generator-nestjs-dto/update-another.dto.ts",
  },
  {
    "content": "
import {IntersectionType} from '@nestjs/swagger'

export class Another {
  id1: string ;
  id2: string ;
}
export class AnotherRel {
  
}
export class AnotherFull extends IntersectionType(Another, AnotherRel) {}
",
    "fileName": "../src/@generated/prisma-generator-nestjs-dto/another.entity.ts",
  },
]
`;
